spec/aws-record/record/marshalers/time_marshaler_spec.rb

```diff
@@ -0,0 +1,125 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+require 'spec_helper'
+require 'time'
+
+module Aws
+  module Record
+    module Marshalers
+      describe TimeMarshaler do
+
+        context 'default settings' do
+          before(:each) do
+            @marshaler = TimeMarshaler.new
+          end
+
+          describe 'type casting' do
+            it 'casts nil and empty string as nil' do
+              expect(@marshaler.type_cast(nil)).to be_nil
+              expect(@marshaler.type_cast('')).to be_nil
+            end
+
+            it 'passes through Time objects' do
+              expected = Time.parse('2015-11-15 17:12:56 +0700')
+              input = Time.parse('2015-11-15 17:12:56 +0700')
+              expect(@marshaler.type_cast(input)).to eq(expected)
+            end
+
+            it 'converts timestamps to Time' do
+              expected = Time.parse("2009-02-13 23:31:30 UTC")
+              input = 1234567890
+              expect(@marshaler.type_cast(input)).to eq(expected)
+            end
+
+            it 'converts Dates to Time' do
+              expected = Time.parse("2009-02-13 08:00:00 UTC")
+              input = Date.parse("2009-02-13")
+              expect(@marshaler.type_cast(input)).to eq(expected)
```

Comment:
This test will only work if the computer running it is in the Pacific Daylight Timezone or equivalent. Needs to be a system time zone agnostic test.